lablib/operators/base.py

```diff
@@ -0,0 +1,7 @@
+class BaseOperator:
+    """Base class for all operators.
+
+    Currently this is only used for type checking.
+    """
+
+    pass
```

Comment:
I'd recommend to make this one `abstract` to enforce that this is not supposed to be instantiated directly.
```suggestion
import abc


class BaseOperator(abc.ABC):
    """Base class for all operators.

    Currently this is only used for type checking.
    """

    pass
```